File: src/fiefapp.entities/Gamesession.cs
using fiefapp.entities.Interfaces;
using System;
using System.Collections.Generic;

namespace fiefapp.entities
{
    public class Gamesession : IBaseEntity
    {
        public Gamesession()
        {
            Id = Guid.NewGuid();
            UserIds = new HashSet<string>();
            PlayerIds = new HashSet<string>();
            Inheritances = new HashSet<Inheritance>();
            Roads = new HashSet<Road>();
            Livingconditions = new HashSet<Livingcondition>();
        }

        public Guid Id { get; set; }
        public string OwnerId { get; set; }
        public ICollection<string> UserIds { get; set; }
        public string GamemasterId { get; set; }
        public ICollection<string> PlayerIds { get; set; }
        public string Name { get; set; }
        public ICollection<Inheritance> Inheritances { get; set; }
        public ICollection<Road> Roads { get; set; }
        public ICollection<Livingcondition> Livingconditions { get; set; }
    }
}

File: src/fiefapp.svelte/src/models/gamesession.ts
import type { Inheritance } from './inheritance';
import type { Livingcondition } from './livingcondition';
import type { Road } from './road';

interface IGamesession {
  id: string;
  ownerId: string;
  userIds: string[];
  gamemasterId: string;
  playerIds: string[];
  name: string;
  inheritances: Inheritance[];
  roads: Road[];
  livingconditions: Livingcondition[];
}

export class Gamesession implements IGamesession {
  id: string;
  ownerId: string;
  userIds: string[];
  gamemasterId: string;
  playerIds: string[];
  name: string;
  inheritances: Inheritance[];
  roads: Road[];
  livingconditions: Livingcondition[];

  constructor() {
    this.id = '';
    this.ownerId = '';
    this.userIds = [];
    this.gamemasterId = '';
    this.playerIds = [];
    this.name = '';
    this.inheritances = [];
    this.roads = [];
    this.livingconditions = [];
  }
}
